test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,118 @@
+#include "test/common/upstream/load_balancer_fuzz_base.h"
+
+#include "test/common/upstream/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+void LoadBalancerFuzzBase::initializeASingleHostSet(uint32_t num_hosts_in_host_set,
+                                                    uint8_t index_of_host_set) {
+  ENVOY_LOG_MISC(trace, "Will attempt to initialize host set {} with {} hosts.", index_of_host_set,
+                 num_hosts_in_host_set);
+  MockHostSet& host_set = *priority_set_.getMockHostSet(index_of_host_set);
+  uint32_t hosts_made = 0;
+  // Cap each host set at 256 hosts for efficiency
+  uint32_t max_num_hosts_in_host_set = 256;
+  // Leave port clause in for future changes
+  while (hosts_made < std::min(num_hosts_in_host_set, max_num_hosts_in_host_set) && port_ < 65535) {
+    host_set.hosts_.push_back(makeTestHost(info_, "tcp://127.0.0.1:" + std::to_string(port_)));
+    ++port_;
+    ++hosts_made;
+  }
+  // TODO: Random call and mod against host_set size for locality logic
+}
+
+// Initializes random and fixed host sets
+void LoadBalancerFuzzBase::initializeLbComponents(
+    const test::common::upstream::LoadBalancerTestCase& input) {
+  random_.initializeSeed(input.seed_for_prng());
+  uint8_t index_of_host_set = 0;
+  for (uint16_t num_hosts_in_host_set : input.num_hosts_in_host_set()) {
+    initializeASingleHostSet(num_hosts_in_host_set, index_of_host_set);
+    index_of_host_set++;
+  }
+  num_host_sets_ = index_of_host_set;
+}
+
+// Updating host sets is shared amongst all the load balancer tests. Since logically, we're just
+// setting the mock priority set to have certain values, and all load balancers interface with host
+// sets and their health statuses, this action maps to all load balancers.
+void LoadBalancerFuzzBase::updateHealthFlagsForAHostSet(uint64_t host_index,
```

Comment:
make sure params are const and/or refs if they are

Reply:
Made all the numeric parameters const.